src/__tests__/theme-provider.test.js

```diff
@@ -0,0 +1,108 @@
+/* eslint func-style:0 */
+import React, {Component} from 'react'
+import {render, mount} from 'enzyme'
+import * as jestGlamorReact from 'jest-glamor-react'
+import toJson from 'enzyme-to-json'
+// eslint-disable-next-line import/default
+import glamorous from '../'
+import ThemeProvider, {CHANNEL} from '../theme-provider'
+
+expect.extend(jestGlamorReact.matcher)
+expect.addSnapshotSerializer(jestGlamorReact.serializer)
+
+const getMockedContext = unsubscribe => ({
+  [CHANNEL]: {
+    getState: () => {},
+    setState: () => {},
+    subscribe: () => unsubscribe,
+  },
+})
+
+test('renders a component with theme', () => {
+  const Comp = glamorous.div(
+    {
+      color: 'red',
+    },
+    (props, theme) => ({padding: theme.padding}),
+  )
+  expect(
+    render(
+      <ThemeProvider theme={{padding: '10px'}}>
+        <Comp />
+      </ThemeProvider>,
+    ),
+  ).toMatchSnapshotWithGlamor()
+})
+
+test('theme properties updates get propagated down the tree', () => {
+  class Parent extends Component {
+    state = {
+      padding: '10px',
+    }
+    setPadding = () => this.setState({padding: '20px'})
+    render() {
+      return (
+        <ThemeProvider theme={{padding: this.state.padding}}>
+          <Child />
+        </ThemeProvider>
+      )
+    }
+  }
+  const Child = glamorous.div(
+    {
+      color: 'red',
+    },
+    (props, theme) => ({padding: theme.padding}),
+  )
+  const wrapper = mount(<Parent />)
+  expect(toJson(wrapper)).toMatchSnapshotWithGlamor(`enzyme.mount`)
+  wrapper.instance().setPadding()
+  expect(toJson(wrapper)).toMatchSnapshotWithGlamor(`enzyme.mount`)
+})
+
+test('merges nested themes', () => {
```

Comment:
Great test. Could we add also the fact that they are truly merged to this test by making the first theme include a `margin: 1`, so we can see that the `margin: 2` from the second theme is what is actually rendered.

Also, note that if your unit is `px` you can make it a number instead of a string. So `{padding: '1px'}` can be: `{padding: 1}` :)